src/test/java/se/nbis/lega/inbox/sftp/S3SftpEventListenerTest.java

```diff
@@ -124,6 +124,52 @@ public void renameFile() throws IOException {
         assertEquals(DigestUtils.sha256Hex(FileUtils.openInputStream(file)), encryptedIntegrity.getChecksum());
     }
 
+    @Test
+    public void renameTopLevelFolder() throws IOException {
+        sftpClient.mkdir("test");
+        sftpClient.mkdir("test/test1");
+
+        sftpClient.put(file.getAbsolutePath(), "test/test1/" + file.getName());
+        fileBlockingQueue.poll();
+
+        sftpClient.rename("test", "test2");
+
+        FileDescriptor fileDescriptor = fileBlockingQueue.poll();
+        assertNotNull(fileDescriptor);
+        assertEquals(username, fileDescriptor.getUser());
```

Comment:
Use Junit rule `ErrorCollector` 
https://carminetech.wordpress.com/2015/10/30/junit-how-to-collect-multiple-assertion-results-in-one-test/